frontend/__tests__/src/components/Metrics/MetricsStep/DeploymentFrequencySettings/PipelineMetricSelection.test.tsx

```diff
@@ -149,13 +149,13 @@ describe('PipelineMetricSelection', () => {
   })
   it('should show no steps warning message when getSteps succeed but get no steps', async () => {
     metricsClient.getSteps = jest.fn().mockReturnValue({ response: [], haveStep: false })
-    const { getByText, getByRole } = await setup(
+    const { getByText, getByRole, getAllByRole } = await setup(
       { id: 0, organization: 'mockOrgName', pipelineName: 'mockName', step: '', branches: [] },
       false,
       false
     )
 
-    await userEvent.click(getByRole('button', { name: PIPELINE_NAME }))
+    await userEvent.click(getAllByRole('button', { name: 'Open' })[1])
```

Comment:
All operation like click should be warped by Act method

Reply:
> All operation like click should be warped by Act method

Thank you for reminding!
Seems like testing-library/react and userEvent already helped us to do this.
https://testing-library.com/docs/preact-testing-library/api/#act